concourse/pipelines/create-cloudfoundry.yml

```diff
@@ -1863,19 +1863,38 @@ jobs:
                     'CF_CLIENT_REDIRECT_URL' => 'https://billing.${SYSTEM_DNS_ZONE_NAME}/oauth/callback',
                     'CF_API_ADDRESS' => '${API_ENDPOINT}',
                   }
-                  manifest = YAML.load_file('manifest.yml')
-                  manifest['applications'].each { |app|
+
+                  api = YAML.load_file('manifest-api.yml')
+                  api['applications'].each { |app|
                     app['env'] = {} unless app['env']
-                    app['env'].merge!(env)
+                    app['env'] = app['env'].merge(env)
                     app['services'] = ['billing-db']
                     app['routes'] = [
                       { 'route' => 'billing.${SYSTEM_DNS_ZONE_NAME}' }
                     ]
                   }
-                  File.write('manifest.yml', manifest.to_yaml)
+                  File.write('manifest-api.yml', api.to_yaml)
+
+                  collector = YAML.load_file('manifest-collector.yml')
+                  collector['applications'].each { |app|
+                    app['env'] = {} unless app['env']
+                    app['env'] = app['env'].merge(env)
+                    app['services'] = ['billing-db']
+                  }
+                  File.write('manifest-collector.yml', collector.to_yaml)
                 "
 
-                cf push paas-billing
+                # FIXME: remove this once all production envs have run this at
+                # least once. It's harmless to leave here, however.
+                cf rename paas-billing paas-billing-collector || true
```

Comment:
Can we do something more like:

```
apps="$(cf apps)"
if echo "${apps}" | grep -qe '^paas-billing ' ; then 
   cf rename paas-billing paas-billing-collector
fi
``` 

That would fail if there is any problem in the call `cf apps` (e.g. network timeout), only do the rename if the app is called `paas-billing` and again fail and stop  if `cf rename` fails.

Reply:
fixed